travis: Fix wasm32 CI

josephlr opened the pull request:
Split emscripten stuff into its own target and stop relying on `cargo web`
to download/manage then emscripten toolchain. We can just get it
ourselves.

We now use `stable` for all WASM32 tests and are now running the emscripten tests.

Fixes #140 

Signed-off-by: Joe Richey <joerichey@google.com>

josephlr commented:
@newpavlov @dhardy dropping the use of `cargo web` for emscripten seems to have fixed the CI.

Other improvements:
  - We now actually run the tests (instead of just building them)
  - The CI is faster (one 6-7 minute stage became two 2-2.5 minute stages)

dhardy commented:
I don't think I can provide any useful feedback here. If someone else (e.g. @zer0x64) can provide feedback then I'll be happy to "approve" (if required).

josephlr commented:
> I don't think I can provide any useful feedback here. If someone else (e.g. @zer0x64) can provide feedback then I'll be happy to "approve" (if required).

@dhardy, I can give some context for the exact steps I added to the CI. I replaced `cargo web prepare-emscripten` with [the steps from the emscripten download page](https://emscripten.org/docs/getting_started/downloads.html). Getting the tests to run was then just a matter of [copying what `cross` does](https://github.com/rust-embedded/cross/blob/master/docker/Dockerfile.asmjs-unknown-emscripten).

Either you or @newpavlov have to approve (I can't approve my own requests).

zer0x64 commented:
If I understand correctly, `cargo-web` is currently broken with `wasm32-unknown-emscripten` as per [this](https://github.com/koute/cargo-web/pull/244), and while the fixed is merge and released, this change aims compile and run directly this specific build without using `cargo web`? 
Also, the CI step is split into two, which is cool because they can now run concurrently and fail separately.
Makes sense to me!
